### DISCUSSION TITLE
q-date range disable weekends

### DISCUSSION THREAD

**Discussion by ikerib** [quasarframework/quasar#14930] [General - Components / Directives / etc] [answered]
Hi,

I want to know if there is any easy way to set weekends disabled for selection. I know I could create a function witch iterates every day and check if it is weekend or no... but... is there any other way? best performance way?

**Comment by metalsadman** [comment, marked as answer]
these examples are already optimized, as options fn are only called on navigation. https://quasar.dev/vue-components/date#example--options.

example https://codepen.io/metalsadman/pen/OJEOxPq open console trigger date navigation to see the perf.

**Comment by ikerib** [comment]
oooh!! Great! I din't know that are called on navigation!
